pkg/hsskey/filereader.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2023 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package hsskey
+
+import (
+	"encoding/binary"
+	"fmt"
+	"hash/crc32"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+const (
+	baseSysfsPattern = "/sys/bus/i2c/devices/%s/eeprom"
+)
+
+// toOctalEscapeSequence converts a byte slice into similar format as C++ protobuf DebugString()
+// If the byte represents a printable ASCII character, add it as is to the output string
+// If it is non-printable or non-ASCII, add it as octal notation.
+func toOctalEscapeSequence(b []byte) string {
+	var shortHandNotions = map[string]string{
+		"\011": "\\t",
+		"\015": "\\r",
+	}
+	octalEscape := ""
+	for _, v := range b {
+		octalChar := fmt.Sprintf("\\%03o", v)
+		if v >= 32 && v <= 126 {
+			octalEscape += string(v)
+		} else if notion, ok := shortHandNotions[string(v)]; ok {
+			// DebugString prints these characters as shorthand notations
+			octalEscape += notion
+		} else {
+			octalEscape += octalChar
+		}
+	}
+	return octalEscape
+}
+
+// validateChecksum check if HSS data is valid based on the checksum value stored in the last 4
+// bytes.
+func validateChecksum(data []byte) bool {
+	size := len(data)
+	if size != hostSecretSeedStructSize {
+		return false
+	}
+	// Split the data into the main part and the checksum. Last 4 bytes is used for storing
+	// checksum value.
+	mainPart := data[:size-4]
+	checksumBytes := data[size-4:]
+
+	// Convert the checksum bytes from big-endian.
```

Comment:
consider commenting on why checksum is stored as big endian order ?  

in addition, `mainPart` sounds like is little endian order.

Reply:
it's based on how the data is serialized. I don't think exactly reason matters much for us as long as we deserialize it correctly.